### PR TITLE
feat: OCA integration

### PR DESCRIPTION
**Pull request by cvarjao**
# Summary of Changes

This PR introduces some initial work towards using [Overlay Capture Architecture (OCA)](http://oca.colossi.network/v1.1.0-rc.html).  The initial support is for the Capture Base (Partial), Label Overlay, and a new (proposed) "Card Layout" overlay.

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: core/App/components/misc/CredentialCard.tsx [hunk 1]
```diff
@@ -32,20 +32,23 @@ const CredentialCard: React.FC<CredentialCardProps> = ({ credential, style = {},
...
-  const overlay = OCABundle.branding[schema]
+  const overlay = bundle?.getCardLayoutOverlay()
+  const metaLayer = bundle?.getMetaOverlay(getCurrentLanguage())
+  const credentialLabel = metaLayer?.name || parsedCredDefName(credential)
+  const credentialBackgroundColor = hashToRGBA(hashCode(credentialLabel))
+  const credentialConnectionLabel = getCredentialConnectionLabel(credential)
 
+  // console.log('overlay header.imageSource:', overlay?.header?.imageSource?.substring(0, 25))
+  // console.log('headerDimensions:', JSON.stringify(headerDimensions))
+  // console.log('headerLogoDimensions:', JSON.stringify(headerLogoDimensions))
```
Comment (wadeking98) [inline core/App/components/misc/CredentialCard.tsx]: should we remove these commented out log statements?

File: core/App/components/misc/CredentialCard.tsx [hunk 2]
```diff
@@ -135,16 +150,14 @@ const CredentialCard: React.FC<CredentialCardProps> = ({ credential, style = {},
...
-            height={headerDimensions.height - paddingVertical}
-            width={headerDimensions.width / 3 - paddingHorizontal}
-            onLayout={({ nativeEvent: { layout } }) => setHeaderLogoDimensions(layout)}
-          />
-        )}
+        <Image
+          source={{ uri: overlay?.header?.imageSource as string }}
+          //height={headerDimensions.height - paddingVertical}
+          //width={headerDimensions.width / 3 - paddingHorizontal}
+          //onLayout={({ nativeEvent: { layout } }) => setHeaderLogoDimensions(layout)}
```
Comment (wadeking98) [inline core/App/components/misc/CredentialCard.tsx]: more commented out lines

File: core/App/screens/CredentialDetails.tsx [hunk 3]
```diff
@@ -97,6 +100,7 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
   }, [])
 
   useEffect(() => {
+    //console.log('Credential:', credential)
```
Comment (wadeking98) [inline core/App/screens/CredentialDetails.tsx]: another console statement to remove

File: core/App/screens/CredentialDetails.tsx [hunk 4]
```diff
@@ -108,6 +112,10 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
     setIsRevoked(isRevoked)
     const isRevokedMessageDismissed = revokedMessageDismissed.has(indyCredentialFormat.credentialRecordId)
     setIsRevokedMessageHidden(isRevokedMessageDismissed)
+    //console.log('OCABundle.getCredentialPresentationFields:')
```
Comment (wadeking98) [inline core/App/screens/CredentialDetails.tsx]: remove old console statement

File: core/__mocks__/custom/@aries-framework_react-hooks.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  const useCredentials = jest.fn().mockReturnValue({ credentials: [] } as any)
+  const useCredentialById = () => {
+    if (credentials !== undefined && credentials.length > 0){
+      //console.log('calling useCredentialById, returning ', credentials)
+      return credentials[0]
+    }
+    //console.log('calling useCredentialById, returning <undefined>')
+    return undefined
+  }
+  // console.log('Mocking ...')
```
Comment (wadeking98) [inline core/__mocks__/custom/@aries-framework_react-hooks.ts]: 3 console statements to remove here

**Review by wadeking98 — COMMENTED** [review]
Looks good, just need to remove the commented out sections

File: core/App/components/misc/CredentialCard.tsx [hunk 6]
```diff
@@ -32,19 +32,19 @@ const CredentialCard: React.FC<CredentialCardProps> = ({ credential, style = {},
...
   const [headerLogoDimensions, setHeaderLogoDimensions] = useState<LayoutRectangle | null>(null)
 
   const { revoked } = state.credential
+  const [bundle, setBundle] = useState<OCACredentialBundle | undefined>(undefined)
 
-  const schema = credentialSchema(credential) || ''
-  const overlay = OCABundle.branding[schema]
+  const overlay = bundle?.getCardLayoutOverlay()
+  const metaLayer = bundle?.getMetaOverlay(getCurrentLanguage())
+  const credentialLabel = metaLayer?.name || parsedCredDefName(credential)
```
Comment (jleach) [inline core/App/components/misc/CredentialCard.tsx]: Should this be `??`?

**Review by jleach — DISMISSED** [review]
One question to consider otherwise LGTM.

**Review by jleach — APPROVED** [review]

**Pull request merged**